money-otel-handler/src/test/java/com/comcast/money/otel/handlers/OtelSpanHandlerSpec.java

```diff
@@ -0,0 +1,173 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.comcast.money.otel.handlers;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.sdk.trace.export.SimpleSpanProcessor;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import io.opentelemetry.trace.SpanContext;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import com.comcast.money.api.SpanId;
+import com.comcast.money.api.SpanInfo;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.never;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({
+        BatchSpanProcessor.class,
+        BatchSpanProcessor.Builder.class,
+        SimpleSpanProcessor.class,
+        SimpleSpanProcessor.Builder.class
+})
+public class OtelSpanHandlerSpec {
+
+    private SpanExporter spanExporter;
+    private BatchSpanProcessor batchSpanProcessor;
+    private BatchSpanProcessor.Builder batchSpanProcessorBuilder;
+    private SimpleSpanProcessor simpleSpanProcessor;
+    private SimpleSpanProcessor.Builder simpleSpanProcessorBuilder;
+
+    private OtelSpanHandler underTest;
+
+    @Before
+    public void beforeEach() throws Exception {
+        PowerMockito.mockStatic(BatchSpanProcessor.class);
+        PowerMockito.mockStatic(SimpleSpanProcessor.class);
+
+        spanExporter = PowerMockito.mock(SpanExporter.class);
+        batchSpanProcessor = PowerMockito.mock(BatchSpanProcessor.class);
+        batchSpanProcessorBuilder = PowerMockito.mock(BatchSpanProcessor.Builder.class);
+        simpleSpanProcessor = PowerMockito.mock(SimpleSpanProcessor.class);
+        simpleSpanProcessorBuilder = PowerMockito.mock(SimpleSpanProcessor.Builder.class);
+
+        PowerMockito.when(BatchSpanProcessor.newBuilder(spanExporter)).thenReturn(batchSpanProcessorBuilder);
+        PowerMockito.when(SimpleSpanProcessor.newBuilder(spanExporter)).thenReturn(simpleSpanProcessorBuilder);
+        PowerMockito.when(batchSpanProcessorBuilder.build()).thenReturn(batchSpanProcessor);
+        PowerMockito.when(simpleSpanProcessorBuilder.build()).thenReturn(simpleSpanProcessor);
```

Comment:
can power mock give false positives in your experience?

Reply:
I haven't had any serious problems with it.  I hate reaching for it when it comes to Unit testing but a lot of the OTel APIs aren't mock-friendly, these builders are all `final` classes.